src/Microsoft.AspNet.Server.Kestrel/Http/NullStream.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Server.Kestrel.Http
+{
+    public class NullStream : Stream
+    {
+        private static Task _emptyTask = Task.FromResult<object>(null);
+
+        public override bool CanRead => false;
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => true;
+
+        public override long Length
+        {
+            get
+            {
+                throw new NotImplementedException();
```

Comment:
Wouldn't it be better to have all operations as noops, in case the implementation of CopyToAsync changes and starts using one of them?